### PR TITLE
Add CommentAdder

### PR DESCRIPTION
**Pull request by Onion1994**

### CODE REVIEW

File: components/CommentAdder.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        .catch(() =>{
+          setIsSendError(true)
+        })
+    }
+
+    return <form onSubmit={(event) => {handleSubmit(event, user, newComment, article)}} className="form-container component">
+        <label htmlFor="add-comment">Add comment</label>
+        <textarea placeholder="Type comment here..." rows="4" id="add-comment" required value={newComment}
+          onChange={(event) => setNewComment(event.target.value)}></textarea>
+            <label htmlFor="add-user">Username</label>
```
Comment (ghost) [inline components/CommentAdder.jsx]: How is someone interacting with your site (say a recruiter) going to know what username will work? I would suggest either one of two things at this point
- a hard coded user and possibly a little tag saying "logged in as xxxx"
- a user in a useContext 

It isn't the best experience to type in a username that only you and I know must exist in the database in order to get a comment to post.

File: components/CommentAdder.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            <input type="text" placeholder="Type username here..." id="add-user" required value={user}
+          onChange={(event) => setUser(event.target.value)} ></input>
+            <button type="submit" className="button blue-button">Send!</button>
+            <p className="error-message">{isSendError ? "Comment couldn't be posted. Please try again." : null}</p>
```
Comment (ghost) [inline components/CommentAdder.jsx]: I like this, and it works well. Good colour contrast, perhaps the text could be a little bigger?

**Review by ghost — CHANGES_REQUESTED** [review]
This is almost there, I would like the adding comments ux to be better. It is really difficult to post a comment at the moment. Your comments don't appear to be optimistically rendering and on a slow connection they take a while to show.

**Review by popatre — CHANGES_REQUESTED** [review]
Screenshot 2023-12-13 at 16 01 51
I couldnt actually review it due to the above error. COuld you sort it and re-submit?

**Review by ghost — APPROVED** [review]
Much better! Good job.

**Pull request merged**